File: emergency-centre-typescript/src/motion-sensor.ts
import Action from "./action";
import Sensor from "./sensor";
import { Vendor } from "./vendor";

export default class MotionSensor extends Sensor {
  private readonly minDistance: number;
  private readonly maxDistance: number;

  constructor(
    id: number,
    vendor: Vendor,
    actions: Action[],
    minDistance: number,
    maxDistance: number
  ) {
    super(id, vendor, actions);
    this.minDistance = minDistance;
    this.maxDistance = maxDistance;
  }

  override toString(): string {
    return `
    MotionSensor{id=${this.getId()}, vendor=${this.getVendor()}, 
    minDistance=${this.minDistance}, maxDistance=${this.maxDistance}}
    `;
  }
}

File: emergency-centre-c#/EmergencyCentre/MotionSensor.cs
using System;

namespace EmergencyCentreDotNet.EmergencyCentre;

public class MotionSensor(int id, Vendor vendor, List<Action> actions, int minDistance, int maxDistance) : Sensor(id, vendor, actions)
{

    public int MinDistance { get; private set; } = minDistance;
    public int MaxDistance { get; private set; } = maxDistance;

    public override bool Equals(object? obj)
    {
        return obj is MotionSensor sensor &&
               base.Equals(obj) &&
               MinDistance == sensor.MinDistance &&
               MaxDistance == sensor.MaxDistance;
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(base.GetHashCode(), MinDistance, MaxDistance);
    }

    public override string ToString()
    {
        return base.ToString() + $"_(minDistance: {MinDistance}, maxDistance: {MaxDistance})";
    }



}
